prow/entrypoint/options.go

```diff
@@ -62,6 +63,13 @@ type Options struct {
 	// Primarily useful in case a subsequent entrypoint will read this entrypoint's marker
 	AlwaysZero bool `json:"always_zero,omitempty"`
 
+	// PropagateErrorCode will cause entrypoint to propagate the error code from its children.
+	// Primarily useful in case you want to exit with a specific error code.
+	PropagateErrorCode bool `json:"propagate_error_code,omitempty"`
+
+	// TestingInterrupt is a channel for tests to interrupt the entrypoint.
+	TestingInterrupt chan os.Signal
```

Comment:
Can we factor this in such a way that testing is done by utilizing the public interface of the code as opposed to requiring custom code paths?

Reply:
Unless there's something explicitly important about doing the test the way you have it, I'd suggest removing this